plugins/glustershd/actor.go

```diff
@@ -0,0 +1,88 @@
+package glustershd
+
+import (
+	"errors"
+
+	"github.com/gluster/glusterd2/glusterd2/daemon"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/glusterd2/xlator"
+        "github.com/gluster/glusterd2/plugins/afr"
+	gderrors "github.com/gluster/glusterd2/pkg/errors"
+)
+
+var names = [...]string{"replicate", "afr"}
+
+type shdActor struct{}
+
+func (actor *shdActor) Do(v *volume.Volinfo, key string, value string) error {
+	glustershDaemon, err := newGlustershd()
+	if err != nil {
+		return err
+	}
+	switch value {
+	case "on":
+		if afr.IsHealEnabled(v) {
```

Comment:
Is this condition required? value is "on"

Reply:
Just an additional check to ensure that "on" has been set in volume options. Only when "on" is set in volume options then proceed to start the daemon.